admin/includes/functions/html_output.php

```diff
@@ -91,6 +91,10 @@ function zen_catalog_href_link($page = '', $parameters = '', $connection = 'NONS
     }
 
     while ( (substr($link, -1) == '&') || (substr($link, -1) == '?') ) $link = substr($link, 0, -1);
+      $link = preg_replace('/(&{2,}|(&amp;)+)/', '&', $link);
+
+      // Convert any remaining '&' into '&amp;' (valid URL for href)
+      $link = str_replace('&', '&amp;', $link);

```

Comment:
Maybe I missed something, but why are you changing the "&" to "&amp;" ?

Reply:
While this was raised as a question in #2806 it should not have been injected in this PR. It should have been done in another branch, specifically PR'd against that Issue ticket.